python/ray/data/_internal/sort.py

```diff
@@ -93,9 +97,18 @@ def sample_boundaries(
     sample_block = cached_remote_fn(_sample_block)
 
     sample_results = [sample_block.remote(block, n_samples, key) for block in blocks]
-    sample_bar = ProgressBar("Sort Sample", len(sample_results))
+
+    should_close_bar = True
+    if ctx is not None and ctx.sub_progress_bar_iter is not None:
+        sample_bar = next(ctx.sub_progress_bar_iter)
+        should_close_bar = False
+    else:
+        sample_bar = ProgressBar("Sort Sample", len(sample_results))
```

Comment:
We can add a TODO to remove the latter case for cleanup later once remove the legacy code.

Reply:
@ericl - I think it's fine not to add TODO here. We plan to delete these stage impl files after moving to optimizer.